bindings/python/src/device.rs

```diff
@@ -0,0 +1,283 @@
+                value: None,
+            },
+            CoreStatus::Occupied(s) => Self {
+                status_type: CoreStatusTypePy::Occupied,
+                value: Some(s),
+            },
+            CoreStatus::Unavailable => Self {
+                status_type: CoreStatusTypePy::Unavailable,
+                value: None,
+            },
+        }
+    }
+}
+
+#[pymethods]
+impl CoreStatusPy {
+    fn __repr__(&self) -> String {
+        match self.status_type {
+            CoreStatusTypePy::Available => String::from("Available"),
+            CoreStatusTypePy::Occupied => format!("Occupied by {}", self.value.as_ref().unwrap()),
+            CoreStatusTypePy::Unavailable => String::from("Unavailable"),
+        }
+    }
+}
+
+#[pyclass(name = "Device", subclass)]
+pub struct DevicePy {
+    pub inner: Arc<Device>,
+}
+
+impl DevicePy {
+    pub fn new(dev: Device) -> Self {
+        Self {
+            inner: Arc::new(dev),
+        }
+    }
+}
+
+#[pymethods]
+impl DevicePy {
+    fn __repr__(&self) -> String {
+        self.inner.to_string()
+    }
+
+    fn name(&self) -> String {
+        self.inner.name()
+    }
+
+    fn device_index(&self) -> u8 {
+        self.inner.device_index()
+    }
+
+    fn arch(&self) -> ArchPy {
+        match self.inner.arch() {
+            Arch::WarboyA0 => ArchPy::WarboyA0,
+            Arch::WarboyB0 => ArchPy::Warboy,
+            Arch::Renegade => ArchPy::Renegade,
+            Arch::U250 => ArchPy::U250,
+        }
+    }
+
+    fn alive(&self) -> PyResult<bool> {
+        self.inner.alive().map_err(to_py_err)
+    }
+
+    fn atr_error(&self) -> PyResult<HashMap<String, u32>> {
+        self.inner.atr_error().map_err(to_py_err)
+    }
+
+    fn busname(&self) -> PyResult<String> {
+        self.inner.busname().map_err(to_py_err)
+    }
+
+    fn pci_dev(&self) -> PyResult<String> {
+        self.inner.pci_dev().map_err(to_py_err)
+    }
+
+    fn device_sn(&self) -> PyResult<String> {
+        self.inner.device_sn().map_err(to_py_err)
+    }
+
+    fn device_uuid(&self) -> PyResult<String> {
+        self.inner.device_uuid().map_err(to_py_err)
+    }
+
+    fn firmware_version(&self) -> PyResult<String> {
+        self.inner.firmware_version().map_err(to_py_err)
+    }
+
+    fn driver_version(&self) -> PyResult<String> {
+        self.inner.driver_version().map_err(to_py_err)
+    }
+
+    fn heartbeat(&self) -> PyResult<u32> {
+        self.inner.heartbeat().map_err(to_py_err)
+    }
+
+    fn ctrl_device_led(&self, led: (bool, bool, bool)) -> PyResult<()> {
+        self.inner.ctrl_device_led(led).map_err(to_py_err)
+    }
+
+    fn core_num(&self) -> u8 {
+        self.inner.core_num()
+    }
+
+    fn cores(&self) -> Vec<u8> {
+        self.inner.cores().to_vec()
+    }
+
+    pub fn dev_files(&self) -> Vec<DeviceFilePy> {
+        self.inner
+            .dev_files()
+            .iter()
+            .map(|d| DeviceFilePy::new(d.clone()))
```

Comment:
```suggestion
            .iter()
            .cloned()
            .map(DeviceFilePy::new)
```

취향 차이인 것 같아서요. 편하실대로 해 주시면 감사하겠습니다.